src/libsystemd/sd-daemon/daemon-util.h

```diff
@@ -0,0 +1,12 @@
+/* SPDX-License-Identifier: LGPL-2.1+ */
+#pragma once
+
+#include <stdbool.h>
+
+#include "sd-daemon.h"
+
+/* This is intended to be used with _cleanup_ attribute. */
+static inline void notify_on_exit(const char **p) {
```

Comment:
Please rename this to `notify_on_cleanup`, or maybe even better `notify_cleanupp`, becuase it's not inherently tied to exiting.